Example/Common/UsingTheAnimation.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  UsingTheAnimation.swift
+//  TheAnimationExample
+//
+//  Created by Carlos on 23/5/18.
+//  Copyright © 2018 marty-suzuki. All rights reserved.
+//
+
+import QuartzCore.CoreAnimation
+import TheAnimation
+
+#if os(iOS) || os(watchOS) || os(tvOS)
+import UIKit
+public typealias Color = UIColor
+public typealias BezierPath = UIBezierPath
+#elseif os(OSX)
+import AppKit
+import CoreGraphics
+public typealias Color = NSColor
+public typealias BezierPath = NSBezierPath
+
+extension NSBezierPath {
+    var cgPath: CGPath {
+        var immutablePath: CGPath
+        let numElements = elementCount
+        let path = CGMutablePath()
+        let points = NSPointArray.allocate(capacity: 3)
+        var didClosePath = true
+        for i in 0..<numElements {
+            switch element(at: i, associatedPoints: points) {
+            case .moveToBezierPathElement:
+                path.move(to: CGPoint(x: points[0].x, y: points[0].y), transform: .identity)
+            case .lineToBezierPathElement:
+                path.addLine(to: CGPoint(x: points[0].x, y: points[0].y), transform: .identity)
+                didClosePath = false
+            case .curveToBezierPathElement:
+                path.addCurve(to: CGPoint(x: points[2].x, y: points[2].y),
+                              control1: CGPoint(x: points[0].x, y: points[0].y),
+                              control2: CGPoint(x: points[1].x, y: points[1].y),
+                              transform: .identity)
+                didClosePath = false
+            case .closePathBezierPathElement:
+                path.closeSubpath()
+                didClosePath = true
+            }
+        }
+        if !didClosePath {
+            path.closeSubpath()
+        }
+        immutablePath = path.copy()!
+        return immutablePath
+    }
+}
+#endif
+
+func performAnimation(view: View, circleLayer: CAShapeLayer) {
```

Comment:
Please write animation codes for each platform's ViewController without summarizing it into one.